tests/SprykerEcoTest/Zed/Braintree/Business/BraintreeFacade/ExecuteCheckoutPostSaveHookTest.php

```diff
@@ -0,0 +1,187 @@
+    {
+        // Assert
+        $this->expectException(RequiredTransferPropertyException::class);
+
+        // Act
+        $this->executeCheckoutPostSaveHookWithEmptyPayment(true);
+    }
+
+    /**
+     * @param bool $isSuccess
+     *
+     * @return \Generated\Shared\Transfer\CheckoutResponseTransfer
+     */
+    protected function executeCheckoutPostSaveHook(bool $isSuccess = true): CheckoutResponseTransfer
+    {
+        /** @var \PHPUnit\Framework\MockObject\MockObject|\SprykerEco\Zed\Braintree\Business\BraintreeBusinessFactory $braintreeBusinessFactoryMock */
+        $braintreeBusinessFactoryMock = $this->getFactoryMock(['createPaymentTransaction', 'createOrderSaver']);
+        $braintreeBusinessFactoryMock->expects($this->once())->method('createPaymentTransaction')->willReturn(
+            $this->getPaymentTransactionMock($isSuccess),
+        );
+        $braintreeBusinessFactoryMock->expects($this->once())->method('createOrderSaver')->willReturn(
+            $this->getOrderSaverMock(),
+        );
+        $braintreeFacade = $this->getBraintreeFacade($braintreeBusinessFactoryMock);
+
+        $orderTransfer = $this->createOrderTransfer();
+        $quoteTransfer = $this->createQuoteTransfer($orderTransfer);
+        $checkoutResponseTransfer = $this->tester->createCheckoutResponseTransfer($orderTransfer);
+
+        return $braintreeFacade->executeCheckoutPostSaveHook($quoteTransfer, $checkoutResponseTransfer);
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\CheckoutResponseTransfer
+     */
+    protected function executeCheckoutPostSaveHookWithEmptyPayment(): CheckoutResponseTransfer
+    {
+        /** @var \PHPUnit\Framework\MockObject\MockObject|\SprykerEco\Zed\Braintree\Business\BraintreeBusinessFactory $braintreeBusinessFactoryMock */
+        $braintreeBusinessFactoryMock = $this->getFactoryMock(['createPaymentTransaction', 'createOrderSaver']);
+        $braintreeBusinessFactoryMock->expects($this->once())->method('createPaymentTransaction')->willReturn(
+            $this->getPaymentTransactionMock(false),
+        );
+        $braintreeBusinessFactoryMock->expects($this->once())->method('createOrderSaver')->willReturn(
+            $this->getOrderSaverMock(),
+        );
+        $braintreeFacade = $this->getBraintreeFacade($braintreeBusinessFactoryMock);
+
+        $orderTransfer = $this->createOrderTransfer();
+        $quoteTransfer = $this->createQuoteTransferWithEmptyPayment($orderTransfer);
+        $checkoutResponseTransfer = $this->tester->createCheckoutResponseTransfer($orderTransfer);
+
+        return $braintreeFacade->executeCheckoutPostSaveHook($quoteTransfer, $checkoutResponseTransfer);
+    }
+
+    /**
+     * @param bool $success
+     *
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerEco\Zed\Braintree\Business\Payment\Transaction\PaymentTransaction
+     */
+    protected function getPaymentTransactionMock(bool $success = true): PaymentTransaction
```

Comment:
same as above

Reply:
Same question here. The ExecuteCheckoutPostSaveHookTest class was moved and github shows it as new, but this particular variable was not introduced in this PR. Should I change the name?

Reply:
Fixed